subnet/kube/kube.go

```diff
@@ -231,7 +233,31 @@ func (ksm *kubeSubnetManager) AcquireLease(ctx context.Context, network string,
 		n.Annotations[backendDataAnnotation] = string(bd)
 		n.Annotations[backendPublicIPAnnotation] = attrs.PublicIP.String()
 		n.Annotations[subnetKubeManagedAnnotation] = "true"
-		n, err = ksm.client.Core().Nodes().Update(n)
+
+		var oldNode, newNode v1.Node
+		if err := api.Scheme.Convert(cacheNode, &oldNode, nil); err != nil {
+			return nil, err
+		}
+		if err := api.Scheme.Convert(n, &newNode, nil); err != nil {
+			return nil, err
+		}
+
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return nil, err
+		}
+
+		newData, err := json.Marshal(newNode)
+		if err != nil {
+			return nil, err
+		}
+
+		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create patch for node %q: %v", ksm.nodeName, err)
+		}
+
+		_, err = ksm.client.Core().Nodes().Patch(ksm.nodeName, api.StrategicMergePatchType, patchBytes, "status")
```

Comment:
@aaronlevy it's a subresource under node that only allows updates to .status fields and annotations. It's less privileged then the node resource and can be acl'd seperately. This code was written pre-rbac and I'm trying to reduce the privilege required by kube-flannel.

https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/node/strategy.go#L122 status strategy zeros out changes to spec.